src/services/codefixes/removeUnnecessaryAwait.ts

```diff
@@ -0,0 +1,32 @@
+/* @internal */
+namespace ts.codefix {
+    const fixId = "removeUnnecessaryAwait";
+    const errorCodes = [
+        Diagnostics.await_has_no_effect_on_the_type_of_this_expression.code,
+    ];
+
+    registerCodeFix({
+        errorCodes,
+        getCodeActions: (context) => {
+            const changes = textChanges.ChangeTracker.with(context, t => makeChange(t, context.sourceFile, context.span));
+            if (changes.length > 0) {
+                return [createCodeFixAction(fixId, changes, Diagnostics.Remove_unnecessary_await, fixId, Diagnostics.Remove_all_unnecessary_uses_of_await)];
+            }
+        },
+        fixIds: [fixId],
+        getAllCodeActions: context => {
+            return codeFixAll(context, errorCodes, (changes, diag) => makeChange(changes, diag.file, diag));
+        },
+    });
+
+    function makeChange(changeTracker: textChanges.ChangeTracker, sourceFile: SourceFile, span: TextSpan) {
+        const awaitKeyword = tryCast(getTokenAtPosition(sourceFile, span.start), (node): node is AwaitKeywordToken => node.kind === SyntaxKind.AwaitKeyword);
+        const awaitExpression = awaitKeyword && tryCast(awaitKeyword.parent, isAwaitExpression);
+        if (!awaitExpression) {
+            return;
+        }
+
+        const parenthesizedExpression = awaitExpression && tryCast(awaitExpression.parent, isParenthesizedExpression);

```

Comment:
what about `(await 1).toFixed()` ? Does this correctly produce `1..toFixed()`?

Reply:
Gooood call. It does not 🙈

Reply:
Is it too embarrassing to leave the parens in and just produce `(1).toFixed()` ? I mean, most people don't like parens, but they've always treated me well.

Reply:
I think that’s what I’ll do, when the inner expression is a DecimalIntegerLiteral.